microsoft/MixedRealityToolkit-Unity
# Fix runtime controller model loading and fix WMRController inheritance

## Overview
This PR adds a base class for WMR sources, allowing us to remove the confusing and constraining inheritance on WMR controller. This also allows us to do controller-specific things, like loading the controller model, without affecting work done by other sources, like loading the hand mesh.

The controller model loading regression happened in #3761, which was done to re-enable hand meshes on HL2. As a result, I felt that fixing both of these issues at the same time made the most sense.

## Changes
- Fixes: #3771, fixes #4342

## Verification
1. Verify that input continues to work on all WMR platforms (HL1, HL2, immersive).
2. Verify that motion controller models load at runtime when deployed to a PC
3. Verify that hand meshes continue to display on HL2
